lib/cocoapods/sources_manager.rb

```diff
@@ -113,9 +113,13 @@ def update(source_name = nil, show_output = false)
         end
 
         changed_spec_paths = {}
-        # Ceate the Spec_Lock file if needed and lock it so that concurrent
+
+        # Do not perform an update if the repos dir has not been setup yet.
+        return unless repos_dir.exist?
+
+        # Create the Spec_Lock file if needed and lock it so that concurrent
         # repo updates do not cause each other to fail
-        File.open("#{Config.instance.repos_dir}/Spec_Lock", File::CREAT) do |f|
+        File.open("#{repos_dir}/Spec_Lock", File::CREAT) do |f|
```

Comment:
Why this change?

Reply:
I figured thats what the `sources_manager` get initialized with. I dont think we will ever have this case but if you had 2 `sources_manager` instances with two different folders they would have to wait for each other for the operation to be completed but they shouldn't have to?